### PR TITLE
Fix incorrect for loop 

### PR DESCRIPTION
**Pull request by lonegunmanb**
As @terrymandin suggested in #19, this pr tried to fix the incorrect for loop.

### CODE REVIEW

**Comment by lonegunmanb** [comment]
Hi @matt-FFFFFF , would you please give this pr a review? Thanks.

**Comment by matt-FFFFFF** [comment]
This looks fine but might be less fragile if we remove lines 10 and 17 and add $d on to the end of the terraform docs command. Also change the relative location of the example configuration.

Thoughts?

@lonegunmanb

**Comment by lonegunmanb** [comment]
@matt-FFFFFF I've extracted the code block into a new function. WDYT?

File: avm_scripts/docs-gen.sh [hunk 1]
```diff
@@ -1,18 +1,21 @@
...
+  local dir=$1
+  echo "===> Generating examples documentation in " $dir
+  rm -f "$dir/.terraform.lock.hcl"
+  if [ -f "$dir/../../.terraform-docs.yml" ]; then
+    terraform-docs -c "$dir/../../.terraform-docs.yml" "$dir"
+  else
+    terraform-docs markdown table --output-file "$dir/README.md" --output-mode inject "$dir"
+  fi
+}
+
```
Comment (matt-FFFFFF) [inline avm_scripts/docs-gen.sh:13]: I think this function should be called from the `examples/` directory. See below as well.

Also the `.terraform-docs.yml` file should be in examples/ not in the module root, as the config differs slightly.

```suggestion
function generate_docs() {
  local dir=$1
  echo "===> Generating examples documentation in " $dir
  rm -f "$dir/.terraform.lock.hcl"
  if [ -f ".terraform-docs.yml" ]; then
    terraform-docs -c ".terraform-docs.yml" "$dir"
  else
    terraform-docs markdown table --output-file "$dir/README.md" --output-mode inject "$dir"
  fi
}

```

File: avm_scripts/docs-gen.sh [hunk 2]
```diff
@@ -1,18 +1,21 @@
...
-  echo "===> Generating examples documentation in " $d
-  cd $d
-  rm -f .terraform.lock.hcl
-  if [ -f "../.terraform-docs.yml" ]; then
-  	terraform-docs -c ../.terraform-docs.yml .
-  else
-  	terraform-docs markdown table --output-file README.md --output-mode inject .
-  fi
-
+  generate_docs $d
```
Comment (matt-FFFFFF) [inline avm_scripts/docs-gen.sh:20]: We should change to the examples directory outside the for loop, alternatively we could use pushd/popd.

```suggestion
cd examples
for d in $examples; do
  generate_docs $d
done
cd ..
```

**Review by matt-FFFFFF — CHANGES_REQUESTED** [review]
What do you think of this suggestion?

File: avm_scripts/docs-gen.sh (hunk 1 above)
Comment (lonegunmanb) [reply avm_scripts/docs-gen.sh:13]: > I think this function should be called from the `examples/` directory. See below as well.
> 
> Also the `.terraform-docs.yml` file should be in examples/ not in the module root, as the config differs slightly.

We might have more config files like Checkov or Tflint configs, for those config files we don't have a better place to put them than the module root, so I think maybe it's better to keep all config files in the root. WDYT? @matt-FFFFFF

Comment (matt-FFFFFF) [reply avm_scripts/docs-gen.sh:13]: The TF docs config for examples is different, so it lives in the examples directory. This is how we have it in the template.

We have 20+ repos all working this way so I think we should leave it.

Do you agree?

**Review by matt-FFFFFF — APPROVED** [review]

**Comment by matt-FFFFFF** [comment]
LGTH :shipit:

**Pull request merged**